src/main/java/study/step2/StringCalculator.java

```diff
@@ -0,0 +1,63 @@
+package study.step2;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private final String CUSTOM_DELIMITER_REGEX = "^//(.)\n";
+    private final String DEFAULT_DELIMITER_REGEX = "[,:]";
+
+    public int sum(String text) {
+        if (isEmpty(text)) {
+            return 0;
+        }
+
+        String[] tokens = split(text);
+        return sum(tokens);
+    }
+
+    private boolean isEmpty(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private String[] split(String text) {
+        String customDelimiter = getCustomDelimiter(text);
+
+        return text.replaceAll(CUSTOM_DELIMITER_REGEX, "").split(customDelimiter);
+    }
+
+    private String getCustomDelimiter(String text) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(text);
```

Comment:
```suggestion
        Matcher matcher = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(text);
```

객체지향 생활체조
규칙 5: 줄여쓰지 않는다(축약 금지).